Update public docs to reflect new name

celinval opened the pull request:
I am updating the documents to avoid using StableMIR and to use `rustc_public` as the crate name. Let me know if that makes sense.

@makai410 / @oli-obk

celinval committed:
Update usage of old stablemir names

celinval committed:
Rename project to "Rustc Librarification"

Rename from "Stable MIR" to better reflect the goal of providing
SemVer compliant APIs rather than completely stable APIs.

makai410 commented on rustc_public/README.md:
```diff
@@ -12,34 +12,35 @@ guide for more information.
 translation between stable and internal constructs will also be done in this crate,
-however, this is currently implemented in the `rustc_smir` crate.[^translation].
-- `rustc_smir`: This crate implements the public APIs to the compiler.
+however, this is currently implemented in the `rustc_public_bridge` crate.[^translation].
```
This isn't the case now, since the translation is already implemented in `rustc_public`. It's okay to do as a follow-up.

celinval replied:
Good point!

celinval replied:
Done!

makai410 reviewed (commented):
Thanks!

celinval committed:
Update README.md

Update the description to reflect the recent refactoring that moves conversion logic to `rustc_public`.
